pkg/xds/envoy/tls.go

```diff
@@ -27,11 +35,17 @@ func CreateDownstreamTlsContext(ctx xds_context.Context, metadata *core_xds.Data
 	}, nil
 }
 
-func CreateUpstreamTlsContext(ctx xds_context.Context, metadata *core_xds.DataplaneMetadata) (*envoy_auth.UpstreamTlsContext, error) {
+// CreateUpstreamTlsContext creates UpstreamTlsContext for outgoing connections
+// It verifies that the client has TLS certificate signed by Mesh CA with URI SAN of spiffe://{mesh_name}/{upstream_service}
```

Comment:
If UpstreamTlsContext is configured on the cluster of the client-side, why it verifies that "client" has TLS certificate? I'm not sure here, but it feels like it verifies "server", doesn't it?